Sources/Plasma/Apps/plClient/Mac-Cocoa/main.mm

```diff
@@ -181,6 +195,8 @@ - (id)init
     self.plsView = view;
     window.contentView = view;
     [window setDelegate:self];
+    
+    gClient.SetClientWindow((__bridge void *)view.layer);
```

Comment:
My hunch was that SetClientDisplay should hold the CGDisplayID or CGDisplayRef - but that's sort of nebulous in since everything you'd do with a display is in flux right now. (Like display modes.) Setting the window value to the NSWindow is an oversight. I _think_ window should be the layer. But I could be talked out of it.

Reply:
On the GL/Linux side, I'm working on introducing a `plClientWindow` class that handles a bunch of the OS-level window stuff, and that works great for Win32 and X11 (and probably Wayland) but AppKit is going to make things difficult again because it's not just C functions... but let's get this merged as-is first and then I'll deal with trying to figure out how to structure things afterwards

Reply:
Yeah - a structure could be a solution. I'm not sure what would need to get passed through. NSWindow comes with the AppKit overhead, which is a problem. (CAMetalLayer has a Metal-cpp front end.)

It may be on the Mac that a CGDisplay and a layer is all that needs to be passed. The layer controls the framebuffer and size. And the CGDisplay can be resolved to a GPU, and a set of display modes.